src/answers-search-bar.js

```diff
@@ -0,0 +1,519 @@
+    const isSuggestQueryTrigger =
+      storage.get(StorageKeys.QUERY_TRIGGER) === QueryTriggers.SUGGEST;
+    if (storage.has(StorageKeys.QUERY) && !isSuggestQueryTrigger) {
+      storage.set(StorageKeys.QUERY_TRIGGER, QueryTriggers.QUERY_PARAMETER);
+    }
+
+    const context = storage.get(StorageKeys.API_CONTEXT);
+    if (context && !isValidContext(context)) {
+      storage.delete(StorageKeys.API_CONTEXT);
+      console.error(`Context parameter "${context}" is invalid, omitting from the search.`);
+    }
+
+    if (storage.get(StorageKeys.REFERRER_PAGE_URL) === undefined) {
+      storage.set(
+        StorageKeys.REFERRER_PAGE_URL,
+        urlWithoutQueryParamsAndHash(document.referrer)
+      );
+    }
+
+    this._services = getServices(parsedConfig, storage);
+
+    this._eligibleForAnalytics = parsedConfig.businessId != null;
+    // TODO(amullings): Initialize with other services
+    if (this._eligibleForAnalytics) {
+      this._analyticsReporterService = new AnalyticsReporter(
+        parsedConfig.experienceKey,
+        parsedConfig.experienceVersion,
+        parsedConfig.businessId,
+        parsedConfig.analyticsOptions,
+        parsedConfig.environment);
+
+      // listen to query id updates
+      storage.registerListener({
+        eventType: 'update',
+        storageKey: StorageKeys.QUERY_ID,
+        callback: id => this._analyticsReporterService.setQueryId(id)
+      });
+
+      this.components.setAnalyticsReporter(this._analyticsReporterService);
+      initScrollListener(this._analyticsReporterService);
+    }
+
+    this.core = new Core({
+      apiKey: parsedConfig.apiKey,
+      storage: storage,
+      experienceKey: parsedConfig.experienceKey,
+      fieldFormatters: parsedConfig.fieldFormatters,
+      experienceVersion: parsedConfig.experienceVersion,
+      locale: parsedConfig.locale,
+      analyticsReporter: this._analyticsReporterService,
+      onVerticalSearch: parsedConfig.onVerticalSearch,
+      onUniversalSearch: parsedConfig.onUniversalSearch,
+      environment: parsedConfig.environment,
+      componentManager: this.components
+    });
+
+    if (parsedConfig.onStateChange && typeof parsedConfig.onStateChange === 'function') {
+      parsedConfig.onStateChange(
+        Object.fromEntries(storage.getAll()),
+        this.core.storage.getCurrentStateUrlMerged());
+    }
+
+    this.components
+      .setCore(this.core)
+      .setRenderer(this.renderer);
+
+    this._setDefaultInitialSearch(parsedConfig.search);
+
+    this.core.init();
+
+    this._onReady = parsedConfig.onReady || function () {};
+
+    this.renderer.init(parsedConfig.templateBundle, this._getInitLocale());
+    this._onReady();
+    if (!this.components.getActiveComponent(SearchComponent.type)) {
+      this._initQueryUpdateListener(parsedConfig.search);
+    }
+    this._searchOnLoad();
+  }
+
+  _initQueryUpdateListener ({ verticalKey, defaultInitialSearch }) {
```

Comment:
we technically don't need the QueryUpdateListener because the redirect searchbar never makes search queries

Reply:
Removed!